wagtail/admin/api/actions/publish.py

```diff
@@ -0,0 +1,59 @@
+from django.core.exceptions import ValidationError as DjangoValidationError
+from django.shortcuts import get_object_or_404
+from rest_framework import fields, status
+from rest_framework.exceptions import ValidationError
+from rest_framework.response import Response
+from rest_framework.serializers import Serializer
+
+from wagtail.api.v2.utils import BadRequestError
+from wagtail.core.actions.publish_page_revision import PublishPageRevisionAction
+
+from .base import APIAction
+
+
+class PublishPageAPIActionSerializer(Serializer):
+    changed = fields.BooleanField(default=True, required=False)
+    log_action = fields.BooleanField(default=True, required=False)
+    previous_revision_id = fields.IntegerField(required=False)
```

Comment:
I think the way the reversion feature was implemented is quite confusing and I think we should make it a separate action later.

If everyone agrees with that. Could we leave it out of the admin API for now?

Reply:
I agree with that. I'll remove it for now waiting to hear others' opinion.